django_cached_field/__init__.py

```diff
@@ -83,9 +88,20 @@ def trigger_cache_recalculation(self):
     obj_name = self._meta.object_name
     if self._meta.proxy:
         obj_name = self._meta.proxy_for_model._meta.object_name
-    offload_cache_recalculation.delay(
-        self._meta.app_label, obj_name, self.pk)
 
+    enqueue_recalculation = offload_cache_recalculation.s(
+        self._meta.app_label,
+        obj_name,
+        self.pk
+    ).delay
+
+    if getattr(settings, CACHED_FIELD_TRANSACTION_AWARE_SETTING, False):
+        from django.db import transaction
+
+        transaction.on_commit(enqueue_recalculation)
```

Comment:
have to make sure that `transaction.on_commit` exists on installed django version